src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.StateSet.cs

```diff
@@ -24,7 +24,7 @@ private class StateSet
 
             // analyzer version this state belong to
             private readonly VersionStamp _analyzerVersion;
-            private readonly AnalyzerTypeData _analyzerTypeData;
+            private readonly AnalyzerStateNames _analyzerStateNames;
```

Comment:
❓ How is this change related to _this_ pull request?

Reply:
while working on this area. saw and changed. 

first thought putting the bool in this AnalyzerTypeData, but decide not to for now, since IsCompilationEndAnalyzer require compilation which is different per project.